lib/form_data.js

```diff
@@ -87,31 +89,40 @@ FormData.prototype._trackLength = function(header, value) {
   });
 };
 
-FormData.prototype._multiPartHeader = function(field, value) {
+FormData.prototype._multiPartHeader = function(field, value, options) {
   var boundary = this.getBoundary();
-  var header =
-    '--' + boundary + FormData.LINE_BREAK +
-    'Content-Disposition: form-data; name="' + field + '"';
-
-  // fs- and request- streams have path property
-  // TODO: Use request's response mime-type
-  if (value.path) {
-    header +=
-      '; filename="' + path.basename(value.path) + '"' + FormData.LINE_BREAK +
-      'Content-Type: ' + mime.lookup(value.path);
-
-  // http response has not
-  } else if (value.readable && value.hasOwnProperty('httpVersion')) {
-    header +=
-      '; filename="' + path.basename(value.client._httpMessage.path) + '"' + FormData.LINE_BREAK +
-      'Content-Type: ' + value.headers['content-type'];
+  var header = '';
+
+  // custom header specified (as string)?
+  // it becomes responsible for boundary
+  // (e.g. to handle extra CRLFs on .NET servers)
+  if (options.header != null) {
```

Comment:
Yes, you're right. This is legit issue. Let me see what I can do. Idea was to freeze new features while packing everything for 1.0 release. I'll get back to you soon.